Lesson4.ipynb

```diff
@@ -29,6 +32,655 @@
+    "\n",
+    "Steps start from column 5, we can count the total steps by looping header index starts from 4, which is column_index > 3\n",
+    "\n",
+    "```python\n",
+    "def totalStepsByRow(row, header):\n",
+    "    totalSteps = 0                                            # define totalSteps\n",
+    "    for header_index, fieldNames in enumerate(header):                   # looping header for counting the steps\n",
+    "        steps_in_column = row[header[header_index]]\n",
+    "        if header_index > 3 and steps_in_column != '':                   # Step amount starts from column 4\n",
+    "            totalSteps = totalSteps + int(steps_in_column)\n",
+    "    return totalSteps\n",
+    "```\n",
+    "** **\n",
+    "By reusing the function, we can get the total steps per person, then compare with 100k/300k/400k\n",
+    "\n",
+    "We can define three string paramters, **NamesForStepsMoreThan100K**, **NamesForStepsMoreThan300K**, **NamesForStepsMoreThan400K** to store the names which satisfy the conditions.\n",
+    "\n",
+    "Please refer to the code block below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Print names of people with steps count more than 100k\n",
+    "import csv\n",
+    "\n",
+    "def totalStepsByRow(row, header):\n",
+    "    totalSteps = 0\n",
+    "    for header_index, fieldNames in enumerate(header):\n",
+    "        steps_in_column = row[header[header_index]]\n",
+    "        if header_index > 3 and steps_in_column != '':\n",
+    "            totalSteps = totalSteps + int(steps_in_column)\n",
+    "    return totalSteps\n",
+    "\n",
+    "with open('Biggest Loser 2018.csv') as csvfile:\n",
+    "    readCSV = csv.DictReader(csvfile)\n",
+    "    header = readCSV.fieldnames\n",
+    "    NamesForStepsMoreThan100K = ''\n",
+    "    \n",
+    "    for row in readCSV:\n",
+    "        totalSteps = sumStepsPerPerson(row, header)\n",
+    "        if totalSteps >= 100000:\n",
+    "            NamesForStepsMoreThan100K = NamesForStepsMoreThan100K + row['team_member'] + ' '\n",
+    "    print(\"Steps more than 100K:\" + NamesForStepsMoreThan100K)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Follow the code block above, can you complete a code block to print the names of people with steps count more than 300k and 400k**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Write you code here:\n",
+    "\n",
+    "# Read from CSV file\n",
+    "# Count for total steps\n",
+    "# Compare with 300K/400k\n",
+    "# Print the names\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Print total steps for each team\n",
+    "\n",
+    "From the code above, we complete a function for counting total steps for each attendee.\n",
```

Comment:
"From the code above, we complete a function" should read "From the code above, we have completed a function"